PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -185,20 +190,28 @@ import Foundation
     }
 
     private class func parseOwnershipType(ownershipType: String?) -> PurchaseOwnershipType {
-        switch ownershipType {
-        case nil:
+        guard let ownershipType = ownershipType else {
+            // TODO: should this be a warning?
             return .purchased
+        }
+
+        switch ownershipType {
         case "PURCHASED":
             return .purchased
         case "FAMILY_SHARED":
             return .familyShared
         default:
-            // TODO(post-migration check): Logging?
+            Logger.warn("received unknown ownershipType: \(ownershipType)")
             return .unknown
         }
     }
 
     private class func parsePeriodType(periodType: String?) -> PeriodType {
```

Comment:
[Style] we should have the `private` stuff in a private extension, right?

Reply:
✅